src/client/automation/playback/type/index.js

```diff
@@ -137,11 +137,8 @@ export default class TypeAutomation {
         if (!this.replace)
             return;
 
-        if (domUtils.isTextEditableElementAndEditingAllowed(this.element)) {
+        if (domUtils.isTextEditableElementAndEditingAllowed(this.element))
```

Comment:
we need some refactoring there. The function is called `_clearText` but it's not truth with your changes (we've discussed it in private).
So I think we should do the following:
```js
_calculateTargetElement () {
        var activeElement     = domUtils.getActiveElement();
        var isContentEditable = domUtils.isContentEditableElement(this.element);

        if (isContentEditable) {
            if (activeElement !== contentEditable.findContentEditableParent(this.element)) {
                this.eventState.skipType = true;

                return;
            }
        }
        else if (activeElement !== this.element) {
            this.eventState.skipType = true;

            return;
        }

        this.element = isContentEditable ? this.element : activeElement;
    }

    _type () {
        if (this.eventState.skipType)
            return Promise.resolve();

        if (this.replace) {
            if (domUtils.isTextEditableElementAndEditingAllowed(this.element))
                textSelection.select(this.element);
            else if (isContentEditable)
                textSelection.deleteSelectionContents(this.element, true);
        }         
        ...
    }
```